src/firstboot/firstboot.c

```diff
@@ -1081,6 +1083,76 @@ static int process_kernel_cmdline(int rfd) {
         return 0;
 }
 
+static int process_reset(int rfd) {
+        _cleanup_close_ int esp_fd = -EBADF, loader_fd = -EBADF;
+        int r;
+
+        if (!arg_reset)
+                return 0;
+
+        FOREACH_STRING(s,
+                       "etc/locale.conf",
+                       "etc/vconsole.conf",
+                       "etc/localtime",
+                       "etc/hostname",
+                       "etc/machine-id",
+                       "etc/kernel/cmdline",
+                       "etc/machine-info",
+                       "var/lib/systemd/random-seed",
+                       "var/lib/systemd/credential.secret") {
+                _cleanup_free_ char *bn = NULL;
+                _cleanup_close_ int pfd = -EBADF;
+
+                r = chase_symlinks_at(rfd, s, CHASE_AT_RESOLVE_IN_ROOT|CHASE_NONEXISTENT|CHASE_PARENT, NULL,
```

Comment:
why CHASSE_NONEXISTENT? if the dir dosn't exist the file won't either, we should handle this gracefully, and immediately go to the next entry on ENOENT